src/input.cpp

```diff
@@ -1373,75 +1546,30 @@ action_id input_context::display_menu( const bool permit_execute_action )
     ctxt.register_action( "COORDINATE" );
     ctxt.register_action( "MOUSE_MOVE" );
     ctxt.register_action( "SELECT" );
-    ctxt.register_action( "SCROLL_UP" );
-    ctxt.register_action( "SCROLL_DOWN" );
-    ctxt.register_action( "UP", to_translation( "Scroll up" ) );
-    ctxt.register_action( "DOWN", to_translation( "Scroll down" ) );
-    ctxt.register_action( "PAGE_DOWN" );
-    ctxt.register_action( "PAGE_UP" );
     ctxt.register_action( "REMOVE" );
     ctxt.register_action( "ADD_LOCAL" );
     ctxt.register_action( "ADD_GLOBAL" );
+    ctxt.register_action( "TEXT.CLEAR" );
+    ctxt.register_action( "PAGE_UP" );
+    ctxt.register_action( "PAGE_DOWN" );
+    ctxt.register_action( "END" );
+    ctxt.register_action( "HOME" );
     if( permit_execute_action ) {
         ctxt.register_action( "EXECUTE" );
     }
     ctxt.register_action( "QUIT" );
-    // String input actions
-    ctxt.register_action( "TEXT.LEFT" );
-    ctxt.register_action( "TEXT.RIGHT" );
-    ctxt.register_action( "TEXT.CLEAR" );
-    ctxt.register_action( "TEXT.BACKSPACE" );
-    ctxt.register_action( "TEXT.HOME" );
-    ctxt.register_action( "TEXT.END" );
-    ctxt.register_action( "TEXT.DELETE" );
-#if defined( TILES )
-    ctxt.register_action( "TEXT.PASTE" );
-#endif
-    ctxt.register_action( "TEXT.INPUT_FROM_FILE" );
```

Comment:
TEXT.PASTE is no longer needed because ImGui supports pasting from clipboard implicitly. TEXT.INPUT_FROM_FILE though is an oversight. I need to think about this a bit. So get_input_string_from_file in input.cpp returns a std::string but it is likely unicode so should I use the helper classes you mentioned before to safely copy it to the char* array?

Reply:
I think it's fine to copy the string directly as long as ImGUI treat strings as utf-8.

Reply:
This is implemented, but there's some problem with the InputText field overruling modifications to the filterText buffer, need to address that separately